clang/lib/CIR/Dialect/Transforms/LibOpt.cpp

```diff
@@ -169,27 +159,41 @@ void LibOptPass::xformStdFindIntoMemchr(StdFindOp findOp) {
 
   CIRBaseBuilderTy builder(getContext());
   builder.setInsertionPointAfter(findOp.getOperation());
-  auto memchrOp0 = builder.createBitcast(
-      iterBegin.getLoc(), iterBegin.getResult(), builder.getVoidPtrTy());
+  auto memchrOp0 =
+      builder.createBitcast(first.getLoc(), first, builder.getVoidPtrTy());
 
   // FIXME: get datalayout based "int" instead of fixed size 4.
-  auto loadPattern = builder.create<LoadOp>(
-      findOp.getOperand(2).getLoc(), underlyingDataTy, findOp.getOperand(2));
+  auto loadPattern =
+      builder.create<LoadOp>(value.getLoc(), underlyingDataTy, value);
   auto memchrOp1 = builder.createIntCast(
       loadPattern, IntType::get(builder.getContext(), 32, true));
 
-  // FIXME: get datalayout based "size_t" instead of fixed size 64.
-  auto uInt64Ty = IntType::get(builder.getContext(), 64, false);
-  auto memchrOp2 = builder.create<ConstantOp>(
-      findOp.getLoc(), uInt64Ty, mlir::cir::IntAttr::get(uInt64Ty, staticSize));
+  const auto uInt64Ty = IntType::get(builder.getContext(), 64, false);
+
+  auto memchrOp2 =
```

Comment:
We should keep the (faster) path that doesn't require a `PtrDiffOp` and add another one that does. Please add another testcase that exercises this more "general" approach.